pixeltable/tests/test_exprs.py

```diff
@@ -356,6 +358,44 @@ def test_arrays(self, test_tbl: catalog.Table) -> None:
         _ = t[t.array_col[:, 0]].show()
         print(_)
 
+    def test_astype(self, test_tbl: catalog.Table) -> None:
+        t = test_tbl
+        # Convert int to float
+        status = t.add_column(c2_as_float=t.c2.astype(FloatType()))
+        assert status.num_excs == 0
+        data = t.select(t.c2, t.c2_as_float).collect()
+        for row in data:
+            assert isinstance(row['c2'], int)
+            assert isinstance(row['c2_as_float'], float)
+            assert row['c2'] == row['c2_as_float']
+
+    def test_apply(self, test_tbl: catalog.Table) -> None:
+        t = test_tbl
+        # For each column c1, ..., c5, we create a new column ci_as_str that converts it to
+        # a string, then check that each row is correctly converted
+        # (For c1 this is the no-op string-to-string conversion)
+        for col_id in range(1, 6):
+            col_name = f'c{col_id}'
+            str_col_name = f'c{col_id}_str'
+            status = t.add_column(**{str_col_name: t[col_name].apply(str)})
+            assert status.num_excs == 0
+            data = t.select(t[col_name], t[str_col_name]).collect()
+            for row in data:
+                assert row[str_col_name] == str(row[col_name])
+        # For columns c6, c7, try using json.dumps and json.loads to emit and parse JSON <-> str
+        for col_id in range(6, 8):
+            col_name = f'c{col_id}'
+            str_col_name = f'c{col_id}_str'
+            back_to_json_col_name = f'c{col_id}_back_to_json'
+            status = t.add_column(**{str_col_name: t[col_name].apply(json.dumps)})
+            assert status.num_excs == 0
+            status = t.add_column(**{back_to_json_col_name: t[str_col_name].apply(json.loads)})
+            assert status.num_excs == 0
+            data = t.select(t[col_name], t[str_col_name], t[back_to_json_col_name]).collect()
+            for row in data:
+                assert row[str_col_name] == json.dumps(row[col_name])
+                assert row[back_to_json_col_name] == row[col_name]
```

Comment:
Missing: failure scenarios

Also, make sure that the tests completely cover the newly-added code (ie, all branches)